packages/react-devtools-scheduling-profiler/src/ImportPage.js

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import type {TimelineEvent} from '@elg/speedscope';
+import type {ReactProfilerData} from './types';
+
+import * as React from 'react';
+import {useEffect, useCallback, useRef} from 'react';
+
+import profilerBrowser from './assets/profilerBrowser.png';
+import style from './ImportPage.css';
+
+import preprocessData from './utils/preprocessData';
+import {readInputData} from './utils/readInputData';
+
+type Props = {|
+  onDataImported: (profilerData: ReactProfilerData) => void,
+|};
+
+export default function ImportPage({onDataImported}: Props) {
+  const processTimeline = useCallback(
+    (events: TimelineEvent[]) => {
+      if (events.length > 0) {
+        onDataImported(preprocessData(events));
+      }
+    },
+    [onDataImported],
+  );
+
+  useEffect(() => {
+    if (__DEV__) {
```

Comment:
> I wonder if we might not be able to do something more clever for DEV like just add a special state to the App component that manages the auto-load (so we don't show the ImportPage UI at all?)

Ooh, great suggestion!

> This could even be URL param driven (in case we want to actually load different profiles, or not load one at all so we could iterate on the ImportPage UI.)

This sounds like something that can be used to support sharing profiles too (https://github.com/MLH-Fellowship/scheduling-profiler-prototype/issues/75)! What I have in mind is a way to allow the user to paste a URL to a raw profile (e.g. the link to a raw GitHub Gist or Paste) and have the profiler load the profile directly from that link. If this link can be specified through a URL param, it can be used both for convenience in DEV and for sharing the profile + profiler with others. I'm not sure if this will be possible due to CORS rules though

Reply:
Yeah, sharing profiles sounds really nice.

There are two ways (at least) that we could do it:
* Pass the URL of a profile JSON that the app could load (may have CORS issues though)
* Pass a URL-encoded string of the actual profile data (maybe too big– but maybe we can compress things, remove bits of the profile we don't care about, etc – or maybe we can export a different format that's more streamlined with only the stuff we need)